// @bit/aureldev.shared-reducer.object-utils/dist/objectUtils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function flattenNextLevel(res, key, val, pre = '') {
    const prefix = [pre, key].filter(v => v).join('.');
    return typeof val === 'object'
        ? Object.keys(val).reduce((prev, curr) => flattenNextLevel(prev, curr, val[curr], prefix), res)
        : Object.assign(res, { [prefix]: val });
}
function flatten(toFlatten) {
    return Object.keys(toFlatten).reduce((prev, curr) => flattenNextLevel(prev, curr, toFlatten[curr]), {});
}
exports.flatten = flatten;
exports.isKeyInArrayObjects = () => { };
exports.isObjectInArray = (array, object) => array.findIndex((element) => element.id === object.id);
exports.updateObjectInArray = (array, object) => {
    const updatedArray = array.map((item) => {
        if (item.id !== object.id) {
            return item;
        }
        return Object.assign(Object.assign({}, item), object);
    });
    return updatedArray;
};
exports.deleteObjectInArray = (array, object) => {
    const index = array.findIndex((element) => element.id === object.id);
    return [...array.slice(0, index), ...array.slice(index + 1)];
};
